main.bal

```diff
@@ -239,7 +245,12 @@ isolated function mapToInternalUrl(string url) returns string {
     // apply all replacements specified in the url map, keys are interpreted as regex
     var replacedUrl = url;
     foreach var [pattern, replacement] in configuredUrlMap.entries() {
-        replacedUrl = regex:replace(replacedUrl, pattern, replacement);
+        do {
+            string:RegExp regex = check regexp:fromString(pattern);
```

Comment:
We need to check in the docker compose if this setting is used with regexes that are no longer compatible. Also this is a breaking that should be mentioned in a changelog/readme of the backend and the qhana-docker repo.

Reply:
The `QHANA_URL_MAPPING` indeed contained a regex that's no longer compatible. I updated the regex. However, I'm not sure if the regex still works as intended.

https://github.com/UST-QuAntiL/qhana-backend/blob/f75fb539d5851c5e43f60b2002c5717d8f02d159/docker-compose.yml#L13

I didn't figure out what the `?<=^` part of the regex is meant for.

Reply:
It is a Positive Look-Behind assertion. You can look regex patterns up using <https://regex101.com> (the old regexes use the Java dialect). It ensures that the part between `(?<=` and `)` matches, but this part is not included in the matched text. This one matches the following:

```text
localhost
^^^^^^^^^

http://localhost:2020
       ^^^^^^^^^
```

But not

```text
ssh://localhost:22
```

Reply:
Unfortunately, Ballerina doesn't support positive lookbehind, yet (https://github.com/ballerina-platform/ballerina-lang/issues/40480). 
Also, I didn't find a way to regex replace with capturing groups.

For now, I spitted the `QHANA_URL_MAPPING` into 2 basic regular expressions:

https://github.com/UST-QuAntiL/qhana-backend/blob/bc9feb76fd7bff26707adea19e23d647681a7717/docker-compose.yml#L13